### PR TITLE
added O365 auth

### PR DESCRIPTION
**Pull request by xiormeesh**
- added controllers to authenticate in Azure and pull the list /me/people for the authenticated user
- o365 config data needs to be added into .env as well as officeLocation used for filtering
- tokens are currently stored in db for simplicity purposes, ideally should be key-value to avoid duplicates
- displaying all users from specified location in the UI in a simple table (saving to db feature is upcoming after mapping file upload is implemented)

### CODE REVIEW

File: app/Http/Controllers/SyncController.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use Microsoft\Graph\Graph;
+use Microsoft\Graph\Model;
+
+class SyncController extends Controller {
+    public function sync() {
+
+        $tokenCache = new TokenCache();
+        $accessToken = $tokenCache->getAccessToken();
+
+        $graph = new Graph();
```
Comment (othercodes) [inline app/Http/Controllers/SyncController.php]: can this be configured as service?? the usage may be easier along the app.

File: composer.json [hunk 2]
```diff
@@ -11,11 +11,13 @@
...
   "require": {
     "php": ">=7.2.0",
+    "ext-libxml": "*",
     "ext-pdo": "*",
-    "ext-sqlite3": "*",
     "ext-simplexml": "*",
-    "ext-libxml": "*",
+    "ext-sqlite3": "*",
     "laravel/lumen-framework": "5.7.*",
+    "league/oauth2-client": "dev-master",
```
Comment (othercodes) [inline composer.json]: not fixed version??

File: app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+use AtlasVG\Http\Controllers\Controller;
+use AtlasVG\TokenStore\TokenCache;
+use Illuminate\Http\Request;
+use Microsoft\Graph\Graph;
+use Microsoft\Graph\Model;
+
+class AuthController extends Controller {
+    public function signin() {
+        
+        $oauthClient = new \League\OAuth2\Client\Provider\GenericProvider([
```
Comment (othercodes) [inline app/Http/Controllers/AuthController.php]: Maybe using service providers will improve the usage of the lib: **https://laravel.com/docs/5.8/providers**

File: app/Http/Controllers/SyncController.php (hunk 1 above)
Comment (xiormeesh) [reply app/Http/Controllers/SyncController.php]: good idea -> future enhancements

File: app/Http/Controllers/AuthController.php (hunk 3 above)
Comment (xiormeesh) [reply app/Http/Controllers/AuthController.php]: good idea -> future enhancements

File: composer.json (hunk 2 above)
Comment (xiormeesh) [reply composer.json]: updated to the latest stable & tested

**Pull request merged**